### DISCUSSION TITLE
Separated duration of cookie consent expiry when Accept necessary and when Accept all is given

### DISCUSSION THREAD

**Discussion by ac-zebrakm** [orestbida/cookieconsent#100] [Ideas]
Hi,

we had a big discussion with lawyers and according to them and also according to Community of privacy data protection in Czech republic is possible to set two different duration periods for Accept all and Accept neccesary.

Our goal is to set up 365 days for Accept all consent. And 30-90 days (shorter is wrong according to Community of privacy data protection discussion) is acceptable time to expire former decission and invoke new question regarding the consent.

So my idea that it would be nice to have to 2 configurations:
{cookie_expiration : 365,
cookie_necessary_only_expiration: 90}

Someone else thinks the same way?

Miro

**Comment by orestbida** [comment]
Hi!

so, in short:

accept all => set cookie for `cookie_expiration` days
accept necessary => set cookie for `cookie_necessary_only_expiration` days

what should happen when a custom selection is made?

**Reply by ac-zebrakm** [reply]
Yes. I am aware of this. I never intended to set particular expiration days per purpose or for custom settings.  So basically 2 modes:
accept (all, custom) => `cookie_expiration` days
accept necessary => `cookie_necessary_only_expiration` days

Everything else look too complicated to me.

**Reply by orestbida** [reply]
Got it. Should be fairly easy to implement.

**Reply by orestbida** [reply]
@lmc-product, this option is now implemented in v2.7.0
